src/components/tiles/table/table-toolbar-buttons.tsx

```diff
@@ -1,14 +1,23 @@
-import React from "react";
+import React, { createContext, useContext } from "react";
 import { Tooltip, TooltipProps } from "react-tippy";
 import classNames from "classnames";
 
 import DeleteSelectedIconSvg from "../../../assets/icons/delete/delete-selection-icon.svg";
 import SetExpressionIconSvg from "../../../clue/assets/icons/table/set-expression-icon.svg";
 import LinkGraphIcon from "../../../clue/assets/icons/table/link-graph-icon.svg";
 import { useTooltipOptions } from "../../../hooks/use-tooltip-options";
+import { useConsumerTileLinking } from "../../../hooks/use-consumer-tile-linking";
+import { getTileDataSet } from "../../../models/shared/shared-data-utils";
+import { TileModelContext } from "../tile-api";
 
 import "./table-toolbar.scss";
 
+export interface IToolbarContext {
+  showExpressionsDialog: () => void;
+  deleteSelected: () => void;
+}
+
+export const ToolbarContext = createContext<IToolbarContext | null>(null);
```

Comment:
My inclination would be to put the context in its own file. More smaller files makes it easier to reduce the import load of modules having to import code they're not using and to avoid dependency cycles, etc.